@benbp travis CI error is the error with version 2.1.0.0. travis CI always to download the latest ansible version, I've locked the version to 2.0.2.0 to make travisCI pass in PR  https://github.com/RackHD/on-imagebuilder/pull/42

log errors:
TASK [setup] *******************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: IndexError: list index out of range
fatal: [/tmp/on-imagebuilder/initrd_build_env/rootfs]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_G43rs8/ansible_module_setup.py\", line 127, in <module>\n    main()\n  File \"/tmp/ansible_G43rs8/ansible_module_setup.py\", line 119, in main\n    data = get_all_facts(module)\n  File \"/tmp/ansible_G43rs8/ansible_modlib.zip/ansible/module_utils/facts.py\", line 3177, in get_all_facts\n  File \"/tmp/ansible_G43rs8/ansible_modlib.zip/ansible/module_utils/facts.py\", line 3123, in ansible_facts\n  File \"/tmp/ansible_G43rs8/ansible_modlib.zip/ansible/module_utils/facts.py\", line 919, in populate\n  File \"/tmp/ansible_G43rs8/ansible_modlib.zip/ansible/module_utils/facts.py\", line 107, in wrapper\n  File \"/tmp/ansible_G43rs8/ansible_modlib.zip/ansible/module_utils/facts.py\", line 1142, in get_mount_facts\nIndexError: list index out of range\n", "module_stdout": "", "msg": "MODULE FAILURE", "parsed": false}

--- a/README.md
+++ b/README.md
@@ -16,7 +16,7 @@ Requirements
 ---------------
 
 - Any Debian/Ubuntu based system (support for other distributions coming soon, theoretically though, just install debootstrap and it should work)
-- Ansible (> 2.0) is installed. To install newest Ansible version ([Ansible Installation Documents](http://docs.ansible.com/ansible/intro_installation.html)):
+- Ansible (> 2.0) (**NOTE**: Version 2.1.0.0 is not workable) is installed. To install newest Ansible version ([Ansible Installation Documents](http://docs.ansible.com/ansible/intro_installation.html)):
 
     ```
     sudo apt-get -y install python-software-properties
@@ -130,6 +130,17 @@ If you have experience with Ansible, some of these steps will be familiar:
     ```
   The wrapper playbooks handle all the setup and cleanup required to run a
   provisioner, such as filesystem mounting and creation, and build file creation.
+- **NOTE**: the customized overlay depends on initrd and basefs build result, that said, before running the overlay playbook, below playbook should run firstly:
+
+    ```
+    sudo ansible-playbook -i hosts common/initrd_wrapper.yml \
+    -e "config_file=vars/initrd.yml provisioner=roles/initrd/provision_initrd"
+
+    sudo ansible-playbook -i hosts common/basefs_wrapper.yml \
+    -e "config_file=vars/basefs.yml provisioner=roles/basefs/provision_rootfs"
+    ```
+
+  so it should run `build_all.sh` which include inirtd and basefs playbook before `build_oem.sh`
 
 ### Changing the global configuration
 

